### DISCUSSION TITLE
Rename INDEX to REF or so?

### DISCUSSION THREAD

**Discussion by stylewarning** [coalton-lang/coalton#936] [Ideas and Proposals]
Can we rename all lookup-like functions to `ref` or `get`? I'm always stumped by `index` because I wonder if it's indexing into or finding the index of something. CL uses `aref`, `svref`, `elt`, `nth`, and `gethash`, so they're not exactly consistent.

A nice thing about `index` currently is that `(index i)` is a good curried accessor.

**Comment by macrologist** [comment]
What about doubling up on index-like accessors? Using `index` for `(index i)` and using `get` for something like `(flip index)` ? 

Sometimes you want to curry over the location and other times over the structure. 

``` lisp

(map (flip vector:index my-vec) (make-list 1 30 400))

;; === 

(map (get my-vec) (make-list 1 30 400)) 

``` 

But its also nice to do 

``` lisp 

(map (vector:index 10) (make-list vec1 vec2 vec3)) 

``` 

I'd favor `position` for `index-of`, if we're nitpicking.

**Reply by stylewarning** [reply]
> What about doubling up on index-like accessors? Using `index` for `(index i)` and using `get` for something like `(flip index)` ? 
> 
> 
> 
> Sometimes you want to curry over the location and other times over the structure. 
> 
> 
> 
> ``` lisp
> 
> 
> 
> (map (flip vector:index my-vec) (make-list 1 30 400))
> 
> 
> 
> ;; === 
> 
> 
> 
> (map (get my-vec) (make-list 1 30 400)) 
> 
> 
> 
> ``` 
> 
> 
> 
> But its also nice to do 
> 
> 
> 
> ``` lisp 
> 
> 
> 
> (map (vector:index 10) (make-list vec1 vec2 vec3)) 
> 
> 
> 
> ``` 
> 
> 
> 
> I'd favor `position` for `index-of`, if we're nitpicking. 

I generally find FLIP distasteful so I'm a fan of your proposition.
